unittest/configurator_test.py

```diff
@@ -23,18 +23,30 @@
 from spinnaker.yaml_util import YamlBindings
 
 
+
 class ConfiguratorTest(unittest.TestCase):
     def test_configuration(self):
-        configurator = Configurator()
-        self.assertEquals(os.path.join(os.environ['HOME'], '.spinnaker'),
-                          configurator.user_config_dir)
+        temp_homedir = tempfile.mkdtemp()
+        home_spinnaker_dir = os.path.join(temp_homedir, '.spinnaker')
+        os.mkdir(home_spinnaker_dir)
+        with open(os.path.join(home_spinnaker_dir,
+                               'spinnaker-local.yml'), 'w') as f:
+          f.write('empty:\n')
+
+        # We arent using HOME in a test, but want to verify that it is
+        # being used.
+        old_home = os.environ['HOME']
+        os.environ['HOME'] = temp_homedir
+        try:
+            configurator = Configurator()
+        finally:
```

Comment:
Subprocess environments shouldnt leak up to the parent's environment.

On Tue, Dec 1, 2015 at 9:46 AM, Lars Wander <notifications@github.com>
wrote:

> In unittest/configurator_test.py
> <https://github.com/spinnaker/spinnaker/pull/636#discussion_r46286001>:
>
> > -        self.assertEquals(os.path.join(os.environ['HOME'], '.spinnaker'),
> > -                          configurator.user_config_dir)
> > +        temp_homedir = tempfile.mkdtemp()
> > +        home_spinnaker_dir = os.path.join(temp_homedir, '.spinnaker')
> > +        os.mkdir(home_spinnaker_dir)
> > +        with open(os.path.join(home_spinnaker_dir,
> > +                               'spinnaker-local.yml'), 'w') as f:
> > +          f.write('empty:\n')
> > +
> > +        # We arent using HOME in a test, but want to verify that it is
> > +        # being used.
> > +        old_home = os.environ['HOME']
> > +        os.environ['HOME'] = temp_homedir
> > +        try:
> > +            configurator = Configurator()
> > +        finally:
>
> Canceling the test with ^C could leave you with a broken home directory.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spinnaker/spinnaker/pull/636/files#r46286001>.
>